Sources/SwiftBuildSupport/PIFBuilder.swift

```diff
@@ -82,12 +146,20 @@ public final class PIFBuilder {
         graph: ModulesGraph,
         parameters: PIFBuilderParameters,
         fileSystem: FileSystem,
-        observabilityScope: ObservabilityScope
+        observabilityScope: ObservabilityScope,
+        pluginScriptRunner: PluginScriptRunner,
+        pluginWorkingDirectory: AbsolutePath,
+        pkgConfigDirectories: [Basics.AbsolutePath],
+        additionalFileRules: [FileRuleDescription]
     ) {
```

Comment:
Consider moving `pluginWorkingDirectory`, `pkgConfigDirectories`, and `additionalFileRules ` to `PIFBuilderParameters` instead. Assuming that doesn't wreak havoc on the calling code ;) 

PS. The `PluginScriptRunner` seems like should be passed on the `init` directly as you just did.